x-pack/plugins/observability/public/pages/rules/components/edit_rule_flyout.tsx

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useMemo, useEffect } from 'react';
+import { useKibana } from '../../../utils/kibana_react';
+import { EditFlyoutProps } from '../types';
+
+export function EditRuleFlyout({ currentRule, onSave }: EditFlyoutProps) {
+  const { triggersActionsUi } = useKibana().services;
+  const [editFlyoutVisible, setEditFlyoutVisibility] = useState<boolean>(false);
+
+  useEffect(() => {
+    setEditFlyoutVisibility(true);
+  }, [currentRule]);
+  const EditAlertFlyout = useMemo(
+    () =>
+      triggersActionsUi.getEditAlertFlyout({
+        initialRule: currentRule,
+        onClose: () => {
+          setEditFlyoutVisibility(false);
+        },
+        onSave,
+      }),
+    [currentRule, setEditFlyoutVisibility, triggersActionsUi, onSave]
+  );
+  return <>{editFlyoutVisible && EditAlertFlyout}</>;
```

Comment:
do you really need the state `editFlyoutVisible`? it looks like that `currentRule` should be sufficient.
```suggestion
  return <>{currentRule && EditAlertFlyout}</>;
```

Reply:
It turns out I still need `editFlyoutVisible`, otherwise flyout stays open when I click the close, cancel or save button.

I do what you suggest in the parent component https://github.com/mgiota/kibana/blob/o11y_rules_page/x-pack/plugins/observability/public/pages/rules/index.tsx#L303

```
{currentRuleToEdit && <EditRuleFlyout onSave={reload} currentRule={currentRuleToEdit} />}
```

I encapsulated the logic of showing/hiding the edit flyout in the EditRuleFlyout component